boa/explorer.py

```diff
@@ -175,20 +185,27 @@ def _fetch(self, **params) -> dict:
 
         return data
 
-    def fetch_abi(self, address: str):
+    def fetch_abi(self, address: str, chain_id: int):
```

Comment:
seems weird that `fetch_abi` can update chain_id. my expectation would be that the only way to change chain_id would be modifying `.chain_id` directly, or even better, to have a setter for it